rcon/commands.py

```diff
@@ -163,7 +220,8 @@ def _timed_request(self, command: str, can_fail=True, log_info=False):
             result=result,
         )
 
-    def _read_list(self, raw):
+    def _read_list(self, raw, conn: HLLConnection):
+        assert conn is not None
```

Comment:
We don't run python with the optimize flag but if we would then that assert statement would be removed. So basically for production code assert should never be used as a mean to raise an exception outside of tests or debug code.

Reply:
good hint, didn't know that for python 🙈 I'll check other ways on how to do that easily :)